src/main/scala/mesosphere/marathon/core/deployment/impl/DeploymentActor.scala

```diff
@@ -164,7 +163,7 @@ private class DeploymentActor(
     }
 
     async {
-      val instances = await(instanceTracker.specInstances(runnableSpec.id))
+      val instances = await(scheduler.getInstances(runnableSpec.id))
       val runningInstances = instances.filter(_.state.condition.isActive)
```

Comment:
should we really pull Active instances here? Shouldn't it be Goal.Running?

Reply:
Good point. This is the orchestration layer. It should only filter for goals and not for conditions.